worker/peergrouper/initiate.go

```diff
@@ -50,16 +45,25 @@ func InitiateMongoServer(p InitiateMongoParams) error {
 
 	// Initiate may fail while mongo is initialising, so we retry until
 	// we successfully populate the replicaset config.
-	var err error
-	for attempt := initiateAttemptStrategy.Start(); attempt.Next(); {
-		err = attemptInitiateMongoServer(p.DialInfo, p.MemberHostPort)
-		if err == nil {
-			logger.Infof("replica set initiated")
-			return err
-		}
-		if attempt.HasNext() {
-			logger.Debugf("replica set initiation failed, will retry: %v", err)
-		}
+	retryCallArgs := retry.CallArgs{
+		Clock:       clock.WallClock,
+		MaxDuration: 60 * time.Second,
+		Delay:       5 * time.Second,
+		Func: func() error {
+			return attemptInitiateMongoServer(p.DialInfo, p.MemberHostPort)
+		},
+		NotifyFunc: func(lastError error, attempt int) {
+			logger.Debugf("replica set initiation failed, will retry: %v", lastError)
```

Comment:
How about using the input attempt? `replica set initiation attempt 1 failed` alludes to the fact that there are multiple attempts, without ever being false.

Reply:
Good idea